api/src/main/java/com/example/api/service/map/ChapterService.java

```diff
@@ -95,4 +96,35 @@ public void deleteChapter(Long chapterID) throws WrongUserTypeException, EntityN
         prevChapter.ifPresent(value -> value.setNextChapter(chapter.getNextChapter()));
         chapterRepo.deleteChapterById(chapterID);
     }
+
+    public void editChapter(EditChapterForm editChapterForm) throws RequestValidationException {
+        // user validation
+        String email = authService.getAuthentication().getName();
+        User professor = userRepo.findUserByEmail(email);
+        userValidator.validateProfessorAccount(professor, email);
+
+        // chapter validation for given editChapterForm
+        Chapter chapter = chapterRepo.findChapterById(editChapterForm.getChapterId());
+        chapterValidator.validateChapterIsNotNull(chapter, editChapterForm.getChapterId());
+        chapterValidator.validateChapterCreation(editChapterForm.getEditionForm());
+        chapterValidator.validateChapterEdition(editChapterForm.getEditionForm(), chapter);
+
+        ChapterForm chapterForm = editChapterForm.getEditionForm();
+
+        // edit basic chapter data
+        chapter.setName(chapterForm.getName());
+        chapter.setPosX(chapterForm.getPosX());
+        chapter.setPosY(chapterForm.getPosY());
+
+        // edit chapter activity map
+        ActivityMap chapterMap = chapter.getActivityMap();
+        chapterMap.setMapSizeX(chapterForm.getSizeX());
+        chapterMap.setMapSizeY(chapterForm.getSizeY());
+        chapterMap.setImage(fileRepo.findFileById(chapterForm.getImageId()));
```

Comment:
Czyli zakładamy, że na froncie przygotujesz request'a w którym podasz aktualne dane rodziału no i część zmienionych z uwagi na edycję?

Reply:
Plan jest taki: biorę komponent na dodanie nowego rozdziału, wypełniam go domyślnie takimi wartościami jakie są obecnie i user może je sobie edytować. Potem wysyła taki set danych

Reply:
Też dlatego nie ma walidacji imageId bo wiemy, że ono będzie poprawne. Chyba że jednak mam dodać jakieś dodatkowe walidacje ?

Reply:
Jeśli na froncie będzie dobrze przepisane to powinno być git, zwłaszcza, że chyba nie ma możliwość usunąć tego obrazu